src/query/src/dist_plan/analyzer.rs

```diff
@@ -44,13 +43,51 @@ impl AnalyzerRule for DistPlannerAnalyzer {
         plan: LogicalPlan,
         _config: &ConfigOptions,
     ) -> datafusion_common::Result<LogicalPlan> {
+        if !Self::is_query(&plan) {
+            return Ok(plan);
+        }
```

Comment:
Then DML with heavy subquery cannot enjoy the rewriting?

Reply:
subquery in dml will be executed separately as a query in `execute_dml_subquery`